tree/tree/src/TLeafF16.cxx

```diff
@@ -218,8 +220,19 @@ void TLeafF16::Streamer(TBuffer &R__b)
    if (R__b.IsReading()) {
       R__b.ReadClassBuffer(TLeafF16::Class(), this);
 
-      if (fTitle.Contains("["))
-	 fElement = new TStreamerElement(Form("%s_Element", fName.Data()), fTitle.Data(), 0, 0, "Float16_t");
+      if (fTitle.Contains("/[")) {
+         auto slash = fTitle.First("/");
+         TString bracket = slash == TString::kNPOS ? fTitle : fTitle(slash + 1, fTitle.Length() - slash - 1);
+         fElement = new TStreamerElement(Form("%s_Element", fName.Data()), bracket.Data(), 0, 0, "Float16_t");
+      } else if (R__b.Class_Version() < 2) {
+         if (fTitle.Contains("[")) {
+            fElement = new TStreamerElement(Form("%s_Element", fName.Data()), fTitle.Data(), 0, 0, "Float16_t");
+            // attempt at rescuing some information lost in fTitle overwrite bug in v1 of this class when saving
+            fTitle = fName + "/" + fTitle;
+         } else {
+            fTitle = fName;
```

Comment:
Did you verify that `fName` is actually containing the array information? (I don't think it does)